daisy/workflow.go

```diff
@@ -793,12 +793,24 @@ func New() *Workflow {
 }
 
 // NewFromFile reads and unmarshals a workflow file.
-// Recursively reads subworkflow steps as well.
-func NewFromFile(file string) (*Workflow, error) {
-	w := New()
+// Recursively reads sub and included steps as well.
+func NewFromFile(file string) (w *Workflow, err error) {
+	w = New()
 	if err := readWorkflow(file, w); err != nil {
 		return nil, err
 	}
+	for _, step := range w.Steps {
```

Comment:
originally, included/sub workflow was read when Validate/populate happens.
Now it happens when NewFromFile was called. The problem is, many attributes were not ready yet. Please ref to "daisycommon.ParseWorkflow": many env settings, for example, OAuth, were populated after NewFromFile is called. Thus, they won't be copied to include/sub workflow.

Reply:
Thanks for the feedback. However, I'm not sure I completely follow the concern. To help me understand, can you write a test that passes with the current behavior on `master`, but fails with this PR?

Reply:
IncludeWorkflow.populate() / SubWorkflow.populate() will copy attributes, for example, OAuth, from its parent workflow.

These attributes were initially populated in daisycommon.ParseWorkflow().
So, if you create Include/sub workflow BEFORE those attributes were populated, they won't affect include/sub workflows.

Reply:
`populate` doesn't happen when the JSON file is loaded; it happens during `Run`. It'd help a lot if you wrote a unit test!

Reply:
I read the CL again and found my misunderstanding. "populate doesn't happen when the JSON file is loaded; it happens during Run" - correct, that is why attributes can be copied to include/sub workflow.

I thought your new condition "&& i.Workflow == nil" will shortcut return. Actually it just skipped loading json. Thanks.